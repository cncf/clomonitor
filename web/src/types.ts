export interface Organization {
  name: string;
  display_name?: string;
  description?: string;
  home_url: string;
  logo_url?: string;
}

export interface BaseProject {
  id: string;
  name: string;
  display_name?: string;
  description?: string;
  home_url: string;
  logo_url?: string;
  devstats_url?: string;
  maturity_id: Maturity;
  category_id: Category;
  score: Score;
  updated_at: number;
}

export interface Project extends BaseProject {
  repositories: BaseRepository[];
  organization: {
    name: string;
  };
}

export interface ProjectDetail extends BaseProject {
  repositories: Repository[];
}

export interface BaseRepository {
  name: string;
  url: string;
  kind: RepositoryKind;
}

export interface Repository extends BaseRepository {
  digest: string;
  repository_id: string;
  score: Score;
  reports: Report[];
}

export interface Report {
  data: CoreReport | any;
  linter_id: LinterId;
  report_id: string;
  updated_at: number;
}

export interface CoreReport {
  // ScoreType
  [key: string]: {
    [key: string]: ReportCheck;
  };
}

export interface ReportCheck {
  passed: boolean;
  value?: string;
  url?: string;
}

export type Score = {
  [key in ScoreType]: number;
} & { score_kind: ScoreKind };

export interface FiltersSection {
  name: string;
  title: string;
  filters: Filter[];
}

export interface Filter {
  name: string | number;
  label: string;
  legend?: string;
  decorator?: JSX.Element;
}

export interface Issue {
  level: number;
  description: string;
}

export interface Prefs {
  search: { limit: number; sort: { by: SortBy; direction: SortDirection } };
  theme: {
    effective: string;
  };
}

export interface ReportOptionData {
  icon: JSX.Element;
  name: string;
  legend: JSX.Element;
}

export enum Maturity {
  Graduated = 0,
  Incubating,
  Sandbox,
}

export enum Category {
  'App definition' = 0,
  Observability,
  Orchestration,
  Platform,
  Provisioning,
  Runtime,
  Serverless,
}

export enum Rating {
  A = 'a',
  B = 'b',
  C = 'c',
  D = 'd',
}

export enum FilterKind {
  Maturity = 'maturity',
  Category = 'category',
  Rating = 'rating',
}

export enum LinterId {
  core = 0,
}

export enum ScoreType {
  BestPractices = 'best_practices',
  Documentation = 'documentation',
  Global = 'global',
  Legal = 'legal',
  License = 'license',
  Security = 'security',
}

export enum SortDirection {
  ASC = 'asc',
  DESC = 'desc',
}

export enum SortBy {
  Name = 'name',
  Score = 'score',
}

export enum RepositoryKind {
  Primary = 'primary',
  Secondary = 'secondary',
}

export enum ScoreKind {
  Primary = 'primary',
  Secondary = 'secondary',
}

export enum ReportOption {
  Adopters = 'adopters',
  ApprovedLicense = 'approved',
  ArtifactHubBadge = 'artifacthub_badge',
  Changelog = 'changelog',
  CodeOfConduct = 'code_of_conduct',
  CommunityMeeting = 'community_meeting',
  Contributing = 'contributing',
  DCO = 'dco',
  Governance = 'governance',
  LicenseScanning = 'scanning',
  Maintainers = 'maintainers',
  OpenSSFBadge = 'openssf_badge',
  Readme = 'readme',
  RecentRelease = 'recent_release',
  Roadmap = 'roadmap',
  SecurityPolicy = 'security_policy',
  SlackPresence = 'slack_presence',
  SPDX = 'spdx_id',
  TrademarkDisclaimer = 'trademark_disclaimer',
  Website = 'website',
}

export interface SearchFiltersURL extends BasicQuery {
  pageNumber: number;
}

export interface BasicQuery {
  text?: string;
  filters?: {
    [key: string]: (string | number)[];
  };
}

export interface SearchQuery extends BasicQuery {
  limit: number;
  offset: number;
  sort_by: SortBy;
  sort_direction: SortDirection;
}

export interface SearchData {
  limit: number;
  offset: number;
  sort_by: string;
  sort_direction: string;
  text?: string;
  category?: number[];
  maturity?: number[];
  rating?: number[];
}

export interface Error {
  kind: ErrorKind;
  message?: string;
}

export enum ErrorKind {
  Other,
  NotFound,
}

export type ReportOptionInfo = {
  [key in ReportOption]: ReportOptionData;
};

export interface RecommendedTemplate {
  name: string;
  url: string;
}
