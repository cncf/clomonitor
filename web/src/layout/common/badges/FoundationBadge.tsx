import { FOUNDATIONS } from '../../../data';
import { Foundation } from '../../../types';
import styles from './Badge.module.css';

interface Props {
  foundation: Foundation;
  className?: string;
}

const FoundationBadge = (props: Props) => {
  const foundationData = FOUNDATIONS[props.foundation];

  return (
    <div
      data-testid="foundation-badge"
      className={`badge text-light extraLightText rounded-0 position-relative ${styles.badge} ${styles.blueBadge} ${props.className}`}
    >
      <div className="d-flex flex-row align-items-center text-uppercase">{foundationData.name}</div>
    </div>
  );
};

export default FoundationBadge;
