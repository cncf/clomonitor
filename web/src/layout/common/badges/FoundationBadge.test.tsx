import { render, screen } from '@testing-library/react';

import { Foundation } from '../../../types';
import FoundationBadge from './FoundationBadge';

describe('FoundationBadge', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<FoundationBadge foundation={Foundation.cncf} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders foundation CNCF', () => {
    render(<FoundationBadge foundation={Foundation.cncf} />);
    expect(screen.getByText('CNCF')).toBeInTheDocument();
  });

  it('renders foundation LFAI', () => {
    render(<FoundationBadge foundation={Foundation.lfaidata} />);
    expect(screen.getByText('LF AI & Data')).toBeInTheDocument();
  });
});
