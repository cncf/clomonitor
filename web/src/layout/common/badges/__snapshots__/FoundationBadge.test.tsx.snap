// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoundationBadge creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="badge text-light extraLightText rounded-0 position-relative badge blueBadge undefined"
    data-testid="foundation-badge"
  >
    <div
      class="d-flex flex-row align-items-center text-uppercase"
    >
      CNCF
    </div>
  </div>
</DocumentFragment>
`;
